### DISCUSSION TITLE
compiling onnxruntime v1.9.0 with onnx v1.9.0 possible?

### DISCUSSION THREAD

**Discussion by Bigpet** [microsoft/onnxruntime#9942] [Other Q&A] [answered]
I'm having trouble getting onnxruntime to compile from the v1.9.0 tag.

There's a bunch of rather fundamental errors, making it seem like the contrib_ops/* source files don't match the rest of the code-base.

Some examples: 
```c++
onnxruntime/core/graph/contrib_ops/nhwc_schema_defs.cc:93:61: error: only virtual member functions can be marked 'override'
  const SparseTensorProto* getInputSparseData(size_t) const override {
                                                            ^~~~~~~~~
```

```c++
onnxruntime/core/graph/contrib_ops/contrib_defs.cc:747:3: error: no viable conversion from 'onnx::OpSchema' to 'onnx::OpSchemaRegistry::OpSchemaRegisterOnce'
  ONNX_CONTRIB_OPERATOR_SCHEMA(FastGelu)
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
``` 

```c++
onnxruntime/core/graph/contrib_ops/contrib_defs.cc:3002:22: error: variable 'numInputs' cannot be implicitly captured in a lambda with no capture-default specified
          } else if (numInputs == 1) {
                     ^
```
                                                            
Is compiling with `onnxruntime_DISABLE_CONTRIB_OPS OFF` not supported for the release?

Edit:
After looking at https://github.com/microsoft/onnxruntime/blob/master/docs/Versioning.md I would assume onnxruntime 1.9.0 should support using onnx 1.9.0 too. I'll take a look if I missed some settings somewhere

**Comment by Bigpet** [comment, marked as answer]
I gave up
onnxruntime 1.8.2 comiles and links with onnx 1.9.0 though

**Comment by snnn** [comment]
Let me clarify Versioning.md a little bit:
First, ONNX Runtime python package doesn't depend on ONNX python package. The table in Versioning.md  means models generated by which ONNX versions can be run with which ONNX Runtime version. For example, ONNX Runtime 1.10 maps to ONNX 1.10 down to 1.2. However, usually ONNX itself doesn't generate model, a model converter does the job.  So the ONNX version range here doesn't mean much.  Even you want to use ONNX 1.2 nowadays, you may not be able to find a way to install it.   The upper limit of the range is a bit more useful.  It means ONNX Runtime was built with a version **equal to or newer** than that. 

ONNX Runtime depends on ONNX's C++ code. One ONNX Runtime commit maps to one ONNX commit. You can find the commit id from ONNX Runtime submodules.  Only that commit is known to work. Others, we don't know and wouldn't pay time to support them. The ONNX commit doesn't have to be an ONNX release commit.  But when we release a package, we will disable the unreleased ONNX operators. For example,  ONNX Runtime 1.10.0 maps to an ONNX commit **equal to or newer than** 1.10.  If any ONNX operator was release after ONNX 1.10, it won't be in ONNX Runtime 1.10.0.

Back to your question: if you want to change the commit id, you will have to patch the code by yourself.  I'm sorry for the inconvenience. I hope it could be more flexible but this is what we have now.
